src/main/java/com/softserveinc/dokazovi/service/impl/AuthorServiceImpl.java

```diff
@@ -0,0 +1,157 @@
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.util.Set;
+
+@Service
+@RequiredArgsConstructor
+public class AuthorServiceImpl implements AuthorService {
+
+	private final DoctorRepository doctorRepository;
+	private final CityRepository cityRepository;
+	private final PasswordEncoder passwordEncoder;
+	private final UserRepository userRepository;
+
+
+	@Override
+	public AuthorDTO save(AuthorDTO authorDTO, UserPrincipal userPrincipal) {
+		if (doctorRepository.findById(authorDTO.getId()).isPresent()) {
+			update(authorDTO, userPrincipal);
+		}
+		AuthorDTO dto;
+		if (userPrincipal.getAuthorities().stream()
+				.anyMatch(
+						grantedAuthority -> grantedAuthority.getAuthority()
+								.equals(RolePermission.CREATE_AUTHOR.getAuthority()))) {
+			DoctorEntity doctorEntity = createDoctorEntity(authorDTO);
+			DoctorEntity doctor = doctorRepository.save(doctorEntity);
+			dto = toAuthorDTO(doctor.getProfile(), doctor);
+			return dto;
+		}
+		throw new ForbiddenPermissionsException();
+	}
+
+	private AuthorDTO toAuthorDTO(UserEntity savedUser, DoctorEntity savedDoctor) {
+		return AuthorDTO.builder().id(savedDoctor.getId())
+				.socialNetwork(savedDoctor.getSocialNetwork())
+				.avatar(savedUser.getAvatar())
+				.bio(savedDoctor.getBio())
+				.firstName(savedUser.getFirstName())
+				.lastName(savedUser.getLastName())
+				.placeOfWork(savedDoctor.getQualification())
+				.email(savedUser.getEmail())
+				.city(savedDoctor.getCity().getId())
+				.build();
+	}
+
+	private DoctorEntity createDoctorEntity(AuthorDTO authorDTO) {
+		CityEntity city = cityRepository.getOne(authorDTO.getCity());
+		return DoctorEntity.builder()
+				.bio(authorDTO.getBio())
+				.qualification(authorDTO.getPlaceOfWork())
+				.socialNetwork(authorDTO.getSocialNetwork())
+				.publishedPosts(0L)
+				.promotionScale(1.0)
+				.profile(toUserEntity(authorDTO))
+				.promotionLevel(UserPromotionLevel.BASIC)
+				.city(city)
+				.build();
+	}
+
+	private UserEntity toUserEntity(AuthorDTO authorDTO) {
+		return UserEntity.builder()
+				.id(null)
+				.firstName(authorDTO.getFirstName())
+				.lastName(authorDTO.getLastName())
+				.email(authorDTO.getEmail())
+				.status(UserStatus.ACTIVE)
+				.enabled(true)
+				.role(RoleEntity.builder()
+						.id(3)
+						.name("Doctor")
+						.permissions(Set.of(RolePermission.SAVE_OWN_PUBLICATION))
+						.build())
+				.password(passwordEncoder.encode("Kolala"))
+				.createdAt(Timestamp.valueOf(LocalDateTime.now()))
+				.build();
+	}
+
+	@Override
+	public AuthorDTO update(AuthorDTO authorDTO, UserPrincipal userPrincipal) {
+		if (userPrincipal.getAuthorities().stream()
+				.anyMatch(
+						grantedAuthority -> grantedAuthority.getAuthority()
+								.equals(RolePermission.UPDATE_AUTHOR.getAuthority()))) {
```

Comment:
I'd suggest creating a universal private method `hasAuthority(UserPrincipal userPrincipal, RolePermission permision)`  that contains the following logic `userPrincipal.getAuthorities().stream()
				.anyMatch(
						grantedAuthority -> grantedAuthority.getAuthority()
								.equals(RolePermission.UPDATE_AUTHOR.getAuthority()))`

Reply:
fixed